gui/wxpython/datacatalog/tree.py

```diff
@@ -893,6 +910,50 @@ def OnSwitchLocationMapset(self, event):
             self.changeLocation.emit(mapset=self.selected_mapset[0].label, location=self.selected_location[0].label)
         self.ExpandCurrentMapset()
 
+    def OnCreateMapset(self, event):
+        """Create new mapset"""
+        self.gisdbase = gisenv()['GISDBASE']
```

Comment:
Don't use self if you don't need the variable outside of the method.

This creates mapset in your current location, not in the selected location.